### PR TITLE
Use config for glibc version

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2443" title="DX-2443" target="_blank">StoryDX-2443</a>  As a user I can specify the preferred glibc version for my machine
  </td></table>
  <br />

### CODE REVIEW

**Review by mitchell-as — CHANGES_REQUESTED** [review]
JIRA says this should be targeted at v43, but the PR is based on v44 and pulling a lot of unnecessary stuff in.

**Comment by MDrakos** [comment]
Whoops. Target & Verify hasn't been working for me for a while so I've just been doing this manually.

File: internal/constants/constants.go [hunk 1]
```diff
@@ -443,6 +443,9 @@ const ReportErrorsConfig = "report.errors"
 // ReportAnalyticsConfig is the config key used to determine if we will send analytics reports
 const ReportAnalyticsConfig = "report.analytics"
 
+// PreferredGlibcVersionConfig is the config key used to determine the preferred glibc version
+const PreferredGlibcVersionConfig = "runimte.preferred.glibc"
```
Comment (mitchell-as) [inline internal/constants/constants.go:447]: There's a typo in this config key name.

File: pkg/platform/model/inventory.go [hunk 2]
```diff
@@ -284,17 +293,17 @@ func filterPlatformIDs(hostPlatform, hostArch string, platformIDs []strfmt.UUID)
...
+func fetchLibcVersion() (string, error) {
 	if runtime.GOOS != "linux" {
-		return ""
+		return "", nil
 	}
 
-	libcInfo := sysinfo.GetRequestedLibcInfo()
-	if libcInfo == nil {
-		return ""
+	cfg, err := config.New()
```
Comment (mitchell-as) [inline pkg/platform/model/inventory.go:301]: Is this really okay? Or do we need to be passing the config instance from main all the way to here?

**Review by mitchell-as — CHANGES_REQUESTED** [review]
This looks mostly fine. I'm guessing this config is system-wide, so a user cannot have one project with one preferred libc version, and another project with the default libc version. Is that something we don't care about supporting?

File: pkg/platform/model/inventory.go (hunk 2 above)
Comment (MDrakos) [reply pkg/platform/model/inventory.go]: Updated to pass the config instance down from main.

**Comment by MDrakos** [comment]
> I'm guessing this config is system-wide, so a user cannot have one project with one preferred libc version, and another project with the default libc version. Is that something we don't care about supporting?

For now, we've opted to go with a "per-machine" config. See comment here: https://activestatef.atlassian.net/browse/DX-2389?focusedCommentId=49153

**Comment by MDrakos** [comment]
Closing in favor of https://github.com/ActiveState/cli/pull/3019

**Pull request closed without merging**